### PR TITLE
Krell spack updates: Add Argo Navis/Krell new gui support library

### PR DESCRIPTION
**Pull request by jgalarowicz**
Add new package for the Argo Navis/Krell new graphical user support library: QtGraph.  It adds convenience routines on top of Qt for use in the new Argo/Krell GUI (cbtf-argonavis-gui) which will be added to spack soon.

### CODE REVIEW

**Comment by adamjstewart** [comment]
Oh boy, another package that builds with QMake, just in time for #4925. Do you want to wait for #4925, or do you want to merge this and I can hack it in #4925? This package looks good aside from a few flake8 errors.

**Comment by jgalarowicz** [comment]
@adamjstewart I can wait.   Looks like I can't have capitol letters in a package name, so I have to redo that.    I have to figure out how to throw away the last commit and start over.

**Comment by adamjstewart** [comment]
@jgalarowicz Are you using `spack create`? It will automatically generate a basic template and choose the correct package name for you.

**Comment by jgalarowicz** [comment]
@adamjstewart I think I did but got confused and changed the name it gave me.  I think 
   git revert <commit number> will remove the last commit and then I can start over.

**Comment by jgalarowicz** [comment]
@adamjstewart Do I need to close the pull request if I remove the changes that I used to create the pull request?  Or just submit a new version of the source changes and this will kick in with Travis checking again.

**Comment by adamjstewart** [comment]
You don't need to close the pull request. You don't use commits to create a pull request, you use branches. As you can see, this branch already has 56 commits on it (you should really create a new branch for each PR).

**Comment by jgalarowicz** [comment]
@adamjstewart Something is fishy the way I'm doing things, because this is a new branch within my fork.
But I keep getting all the previous commits for some reason?
This is the sequence I'm using.
git checkout -t origin/develop
git remote add upstream https://github.com/LLNL/spack.git
git fetch upstream
git merge upstream/develop
then
git checkout somename
git add package.py
git commit -m "..." package.py
git push origin somename

**Comment by adamjstewart** [comment]
Try `git pull -f upstream develop`. I have a feeling you accidentally committed some changes to your local develop branch. Every time you do a fetch/merge, you are adding to your local history, making it increasingly divergent from upstream develop. `git pull -f` will force overwrite your develop branch with upstream develop.

**Comment by jgalarowicz** [comment]
@adamjstewart Do I do the  git pull -f upstream develop in place of one of the commands in my list posted above?  Thanks for your help!

**Comment by adamjstewart** [comment]
I would do:
```
$ git checkout develop          # switch to the develop branch
$ git pull -f upstream develop  # wipe out any local changes with upstream
$ git push -f origin develop    # wipe out any remote changes with upstream
```
This will get you back in line. Then, from now on, you can use:
```
$ git checkout develop       # switch to the develop branch
$ git pull upstream develop  # update local develop branch
$ git push origin develop    # update remove develop branch
```
to update, and:
```
$ git checkout develop          # switch to the develop branch
$ git branch features/foo       # create a features/foo branch off of develop
$ git checkout features/foo     # switch to this new branch
$ git add package.py            # add changes to the stage
$ git commit -m "message"       # commit the changes
$ git push origin features/foo  # push these changes to your remote
```
to submit a PR.

**Comment by adamjstewart** [comment]
Okay, #4925 has been merged now. I think the only changes necessary will be to replace `Package` with `QMakePackage` and drop your `install()` method. You might as well blow away your develop branch away and start from a fresh copy.

**Comment by jgalarowicz** [comment]
@adamjstewart Thanks!  I'm working on this.

File: var/spack/repos/builtin/packages/qtgraph/package.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            libdir = self.prefix.lib
+
+        # The implementor has set up the library and include paths in
+        # a non-conventional way.  We reflect that here.
+        run_env.prepend_path(
+            'LD_LIBRARY_PATH', join_path(
+                libdir, '{0}'.format(self.spec['qt'].version.up_to(3))))
+
+        run_env.prepend_path(
+            'CPATH', join_path(self.prefix, 'include', 'QtGraph'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/qtgraph/package.py]: You can use `self.prefix.include.QtGraph` now.

File: var/spack/repos/builtin/packages/qtgraph/package.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/qtgraph/package.py]: This blank line at the end of the file will cause the flake8 tests to fail.

File: var/spack/repos/builtin/packages/qtgraph/package.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        spack_env.set('GRAPHVIZ_ROOT', self.spec['graphviz'].prefix)
+        spack_env.set('INSTALL_ROOT', self.prefix)
+
+        # The assumed qt version changed to QT5 (as of paraview 5.2.1),
+        # so explicitly specify which QT major version is actually being used
+
+        # What lib suffix do we need to use?
+        libdir = self.prefix.lib64
+        if not os.path.isdir(libdir):
+            libdir = self.prefix.lib
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/qtgraph/package.py]: I prefer `prefix.lib`, but it depends on where the package decides to install to. In any case, it probably isn't dynamic. See which one it is using and stick with it.

File: var/spack/repos/builtin/packages/qtgraph/package.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+class Qtgraph(QMakePackage):
+    """The baseline library used in the CUDA-centric Open|SpeedShop Graphical
+       User Interface (GUI) which allows Graphviz DOT formatted data to be
+       imported into a Qt application by wrapping the Graphviz libcgraph and
+       libgvc within the Qt Graphics View Framework."""
+
+    # url for qtgraph package's homepage here.
+    homepage = "https://github.com/OpenSpeedShop/QtGraph"
+
+    # Use when the git repository is available
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/qtgraph/package.py]: It looks like the git repo is now available. Probably don't need this comment anymore.

File: var/spack/repos/builtin/packages/qtgraph/package.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+import os
+
+
+class Qtgraph(QMakePackage):
+    """The baseline library used in the CUDA-centric Open|SpeedShop Graphical
+       User Interface (GUI) which allows Graphviz DOT formatted data to be
+       imported into a Qt application by wrapping the Graphviz libcgraph and
+       libgvc within the Qt Graphics View Framework."""
+
+    # url for qtgraph package's homepage here.
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/qtgraph/package.py]: This is implied by the fact that we are in the `qtgraph` package and the variable is called `homepage`. I don't think this comment adds much.

**Pull request merged**